lib/spack/docs/getting_started.rst

```diff
@@ -57,3 +112,928 @@ This will install a new spack script in ``/my/favorite/prefix/bin``,
+              cxx: /opt/intel-15.0.24/bin/icpc-15.0.24-beta
+              f77: /opt/intel-15.0.24/bin/ifort-15.0.24-beta
+              fc: /opt/intel-15.0.24/bin/ifort-15.0.24-beta
+            spec: intel@15.0.24.4.9.3
+
+
+    .. note::
+
+        The version number on the Intel compiler is a combination of
+        the "native" Intel version number and the GNU compiler it is
+        targeting.
+
+    .. warning::
+
+        This solution has not yet been tested.  Details may vary.
+
+""""""""""""""""""""""""""
+Command Line Configuration
+""""""""""""""""""""""""""
+
+. warning::
+
+    As of the writing of this manual, added compilers flags are broken;
+    see `GitHub Issue <https://github.com/LLNL/spack/pull/1532>`_.
+
+One can also control which GCC is seen by the Intel compiler by adding
+flags to the ``icc`` command:
+
+ #. Identify the location of the compiler you just installed:
+
+    .. code-block:: console
+
+        $ spack location -i gcc
+        /home2/rpfische/spack2/opt/spack/linux-centos7-x86_64/gcc-4.9.3-iy4rw...
+
+#. Set up ``compilers.yaml``, for example:
+
+    .. code-block:: yaml
+
+        compilers:
+        - compiler:
+            modules = [intel-15.0.24]
+            operating_system: centos7
+            paths:
+              cc: /opt/intel-15.0.24/bin/icc-15.0.24-beta
+              cflags: -gcc-name /home2/rpfische/spack2/opt/spack/linux-centos7-x86_64/gcc-4.9.3-iy4rw.../bin/gcc
+              cxx: /opt/intel-15.0.24/bin/icpc-15.0.24-beta
+              cxxflags: -gxx-name /home2/rpfische/spack2/opt/spack/linux-centos7-x86_64/gcc-4.9.3-iy4rw.../bin/g++
+              f77: /opt/intel-15.0.24/bin/ifort-15.0.24-beta
+              fc: /opt/intel-15.0.24/bin/ifort-15.0.24-beta
+              fflags: -gcc-name /home2/rpfische/spack2/opt/spack/linux-centos7-x86_64/gcc-4.9.3-iy4rw.../bin/gcc
+            spec: intel@15.0.24.4.9.3
+
+.. warning::
+
+    This solution has not yet been tested.  Details may vary.
+
+
+^^^
+NAG
+^^^
+
+At this point, the NAG compiler is `known to not
+work<https://github.com/LLNL/spack/issues/590>`.
+
+
+^^^
+PGI
+^^^
+
+PGI comes with two sets of comilers for C++ and Fortran,
+distinguishable by their names.  "Old" compilers:
+
+.. code-block:: yaml
+
+    cc:  /soft/pgi/15.10/linux86-64/15.10/bin/pgcc
+    cxx: /soft/pgi/15.10/linux86-64/15.10/bin/pgCC
+    f77: /soft/pgi/15.10/linux86-64/15.10/bin/pgf77
+    fc:  /soft/pgi/15.10/linux86-64/15.10/bin/pgf90
+
+"New" compilers:
+
+.. code-block:: yaml
+
+    cc:  /soft/pgi/15.10/linux86-64/15.10/bin/pgcc
+    cxx: /soft/pgi/15.10/linux86-64/15.10/bin/pgc++
+    f77: /soft/pgi/15.10/linux86-64/15.10/bin/pgfortran
+    fc:  /soft/pgi/15.10/linux86-64/15.10/bin/pgfortran
+
+Older installations of PGI contains just the old compilers; whereas
+newer installations contain the old and the new.  The new compiler is
+considered preferable, as there are results that some packages
+(``hdf4``) will not build with the old compiler.
```

Comment:
For the record, I believe the problem with HDF4 is that everything is compiled with the F77 compiler, but at some point some F90 code slipped in there. So compilers that can handle both F77 and F90 (gfortran, pgfortran, etc) are fine, but compilers specific to one or the other (pgf77, pgf90) won't work. Probably don't need to add this to the documentation though.